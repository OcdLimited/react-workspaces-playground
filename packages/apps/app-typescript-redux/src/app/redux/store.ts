import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import rootReducer from './rootReducer';
import rootSaga from './sagas/sagas';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

const sagaMiddleware = createSagaMiddleware();

export const store = {
	...configureStore({
		reducer: rootReducer,
		enhancers: enhancers,
		middleware: [sagaMiddleware],
	}),
	injectedSagas: [],
	runSaga: sagaMiddleware.run,
};

sagaMiddleware.run(rootSaga);

if (process.env.NODE_ENV === 'development' && module.hot) {
	module.hot.accept('./rootReducer', () => {
		const newRootReducer = require('./rootReducer').default;
		store.replaceReducer(newRootReducer);
	});
}

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
export type AppDispatch = typeof store.dispatch;

export default store;
