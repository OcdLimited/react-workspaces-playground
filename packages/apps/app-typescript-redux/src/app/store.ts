import { configureStore, ThunkAction, Action, StoreEnhancer } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import counterReducer from '../features/counter/counterSlice';

const enhancers = new Array<StoreEnhancer>();

if (process.env.NODE_ENV !== 'production') {
  enhancers.push(withReduxEnhancer);
}

export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
  enhancers: enhancers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
