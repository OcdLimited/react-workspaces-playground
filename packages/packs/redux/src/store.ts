import { configureStore, StoreEnhancer, Store, AnyAction } from '@reduxjs/toolkit';
import withReduxEnhancer from 'addon-redux/enhancer';
import createSagaMiddleware from 'redux-saga';
import { createApiMiddleware, appConfigSaga } from '@ocdlimited/abp.react.core';
import { createRootReducer } from './reducers/rootReducer';

export interface InjectableStore extends Store<any, AnyAction> {
	injectedReducers: any;
	injectedSagas: any;
	runSaga(saga: any, args: any): any;
}

const enhancers: StoreEnhancer[] = [];

if (process.env.NODE_ENV !== 'production') {
	enhancers.push(withReduxEnhancer);
}

export function buildStore(rootReducer: any, preloadedState?: any): InjectableStore {
	const sagaMiddleware = createSagaMiddleware();

	const store = {
		...configureStore({
			reducer: createRootReducer(rootReducer),
			enhancers: enhancers,
			middleware: [createApiMiddleware(), sagaMiddleware],
			preloadedState,
		}),
		injectedSagas: [],
		injectedReducers: { ...rootReducer },
		runSaga: sagaMiddleware.run,
	};

	sagaMiddleware.run(appConfigSaga);

	/* istanbul ignore next */
	if (process.env.NODE_ENV === 'development' && module.hot) {
		if (module.hot) {
			// Enable Webpack hot module replacement for reducers
			module.hot.accept('./reducers', () => {
				const nextReducer = createRootReducer(store.injectedReducers);
				store.replaceReducer(nextReducer);
			});
		}
	}

	return store;
}
