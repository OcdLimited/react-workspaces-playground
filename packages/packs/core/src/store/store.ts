import { combineReducers } from '@reduxjs/toolkit';
import { InjectableStore, buildStore as innerBuildStore } from '@ocdlimited/abp.react.redux';
import { apiReducer, createApiMiddleware } from '../api';
import { appConfigReducer, appConfigSaga } from '../application-configuration';
import { toastReducer } from '../toast';
import { tokenReducer } from '../token';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function createRootReducer(injectedReducers: any): any {
	return combineReducers({
		api: apiReducer,
		config: appConfigReducer,
		toast: toastReducer,
		token: tokenReducer,
		...injectedReducers,
	});
}

export function buildStore(rootReducer: unknown, preloadedState?: unknown): InjectableStore {
	const reducer = createRootReducer(rootReducer);
	return innerBuildStore(reducer, createRootReducer, preloadedState, [createApiMiddleware()], [appConfigSaga]);
}

export default buildStore;
