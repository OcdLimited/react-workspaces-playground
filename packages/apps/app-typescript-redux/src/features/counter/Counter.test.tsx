import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../app/store';
import { Counter } from './Counter';

test('increment', () => {
	const { getByText } = render(
		<Provider store={store}>
			<Counter />
		</Provider>,
	);
	const increment = getByText(/\+/i);
	expect(increment).toBeInTheDocument();
	fireEvent.click(increment);
});

test('decrement', () => {
	const { getByText } = render(
		<Provider store={store}>
			<Counter />
		</Provider>,
	);
	const increment = getByText(/-/i);
	expect(increment).toBeInTheDocument();
	fireEvent.click(increment);
});

test('Add Amount', () => {
	const { getByText } = render(
		<Provider store={store}>
			<Counter />
		</Provider>,
	);
	const add = getByText(/Add Amount/i);
	expect(add).toBeInTheDocument();
	fireEvent.click(add);
});

test('Add Async', () => {
	const { getByText } = render(
		<Provider store={store}>
			<Counter />
		</Provider>,
	);
	const add = getByText(/Add Async/i);
	expect(add).toBeInTheDocument();
	fireEvent.click(add);
});

test('Set increment amount', () => {
	const { getByLabelText } = render(
		<Provider store={store}>
			<Counter />
		</Provider>,
	);
	const add = getByLabelText(/Set increment amount/i);
	expect(add).toBeInTheDocument();
	fireEvent.change(add, { target: { value: 'a' } });
});
