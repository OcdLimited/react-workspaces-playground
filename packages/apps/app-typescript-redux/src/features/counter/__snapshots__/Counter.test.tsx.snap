// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <button
        aria-label="Increment value"
        class="button"
      >
        +
      </button>
      <span
        class="value"
      >
        2
      </span>
      <button
        aria-label="Decrement value"
        class="button"
      >
        -
      </button>
    </div>
    <div
      class="row"
    >
      <input
        aria-label="Set increment amount"
        class="textbox"
        value="2"
      />
      <button
        class="button"
      >
        Add Amount
      </button>
      <button
        class="asyncButton"
      >
        Add Async
      </button>
    </div>
  </div>
</div>
`;
